tests/test_bitbucket_server.py

```diff
@@ -0,0 +1,248 @@
+from unittest import TestCase
+
+from mock import patch, Mock, MagicMock
+from stashy.errors import NotFoundException, GenericException
+
+from pyup.errors import BranchExistsError, RepoDoesNotExistError
+from pyup.providers.bitbucket_server import Provider, BadTokenError, BadRepoNameError
+from pyup.requirements import RequirementsBundle
+
+
+class ProviderTest(TestCase):
+    def setUp(self):
+        self.provider = Provider(bundle=Mock())
+        self.provider._api = Mock()
+        self.repo = Mock()
+        self.token = "foo@foo@http://example.com/stash"
```

Comment:
I still think the `base_url` looks weird. Can we make something like this instead?

```python
provider = Provider(bundle=Mock(), url="http://example.com/stash")
token = "foo@bar"
```